llvm/include/llvm/Config/llvm-config-enable-curl.h.cmake

```diff
@@ -0,0 +1,16 @@
+/*===------- llvm/Config/llvm-config-enable-curl.h.cmake ----------*- C -*-===*/
+/*                                                                            */
+/* Part of the LLVM Project, under the Apache License v2.0 with LLVM          */
+/* Exceptions.                                                                */
+/* See https://llvm.org/LICENSE.txt for license information.                  */
+/* SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception                    */
+/*                                                                            */
+/*===----------------------------------------------------------------------===*/
+
+#ifndef LLVM_CONFIG_ENABLE_CURL_H
+#define LLVM_CONFIG_ENABLE_CURL_H
+
+/* Define if we have curl and want to use it */
+#cmakedefine LLVM_ENABLE_CURL ${LLVM_ENABLE_CURL}
```

Comment:
I worry that this split is going to cause subtle bugs if we don't use cmakedefine01 everywhere. Otherwise we could end up with #ifndef being evaluated incorrectly.

Reply:
At the moment we keep including everything through the llvm-config.h ; but you're right this is a major hurdle.

I'm already going one by one through all the options and converting them to cmakedefine01.
I am also making the LLVM codebase `-Wundef` clean: this is the only way to catch this in a principle way (I think we should enable -Wundef by default actually).

Reply:
I had a long flight yesterday, I have locally all the patches for making every option a `cmakedefine01` and making the codebase -Wundef clean.